include/ghex/transport_layer/util/barrier.hpp

```diff
@@ -128,7 +127,8 @@ namespace gridtools {
                 void in_node2() const
                 {
                     size_t ex = b_count2;
-                    while(!b_count2.compare_exchange_weak(ex, ex-1, std::memory_order_relaxed)) ex = b_count2;
+                    while(!b_count2.compare_exchange_weak(ex, ex-1, std::memory_order_relaxed))
+                        ex = b_count2;
                     if (ex == 1) {
```

Comment:
@mbianco Should this not be `ex == 0` ?

Reply:
Also, in `in_node1` it should read `expected == m_threads` instead of `expected == m_threads-1`

Reply:
@angainor I think `ex==1` is correct here. same goes for `in_node1`. @mbianco can you comment?